src/extractor/guidance/turn_handler.cpp

```diff
@@ -78,14 +78,32 @@ namespace guidance
 {
 
 // a wrapper to handle road indices of forks at intersections
-TurnHandler::Fork::Fork(const Intersection::iterator right, const Intersection::iterator left)
-    : right(right), left(left), size((left - right) + 1)
+TurnHandler::Fork::Fork(const Intersection::iterator intersection_base,
+                        const Intersection::iterator begin,
+                        const Intersection::iterator end)
+    : intersection_base(intersection_base), begin(begin), end(end), size(end - begin)
 {
-    BOOST_ASSERT(right < left);
-    BOOST_ASSERT(size >= 2);
-    BOOST_ASSERT(size <= 3);
+    BOOST_ASSERT(begin < end);
+    BOOST_ASSERT(size == 2 || size == 3);
 }
 
+ConnectedRoad &TurnHandler::Fork::getRight() { return *begin; }
+ConnectedRoad &TurnHandler::Fork::getLeft() { return *(end - 1); }
+ConnectedRoad &TurnHandler::Fork::getMiddle()
+{
+    BOOST_ASSERT(size == 3);
+    return *(begin + 1);
+}
+ConnectedRoad &TurnHandler::Fork::getRight() const { return *begin; }
+ConnectedRoad &TurnHandler::Fork::getLeft() const { return *(end - 1); }
```

Comment:
The Fork constructor makes sure this doesn't happen (`BOOST_ASSERT(begin < end);`)